android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageView.java

```diff
@@ -127,4 +195,117 @@ else if (seconds > 60) {
         }
         return result;
     }
+
+    private void showBackgroundImage() {
+        ViewGroup mNewTabPageLayout = getNewTabPageLayout();
+        TextView creditText = (TextView)mNewTabPageLayout.findViewById(R.id.credit_text);
+
+        if(mSharedPreferences.getBoolean(BackgroundImagesPreferences.PREF_SHOW_BACKGROUND_IMAGES, true)
+            && Build.VERSION.SDK_INT > Build.VERSION_CODES.M) {
+            ViewTreeObserver observer = mNewTabPageLayout.getViewTreeObserver();
+            observer.addOnGlobalLayoutListener(new ViewTreeObserver.OnGlobalLayoutListener() {
```

Comment:
this seems like a lot of code to dump into BraveNewTabPageView.java, can we create an actual listener subclass instead of defining one inside this method?